types/lodash/common/object.d.ts

```diff
@@ -1021,8 +1021,41 @@ declare module "../index" {
         functionsIn(): CollectionChain<string>;
     }
 
-    type GetIndexedField<T, K> = K extends keyof T
-        ? T[K]
+    /**
+     * If a value of type T can be indexed by a key of type K
+     * GetIndexedField<T, K> deduces the type of value at that key,
+     * otherwise it returns undefined type.
+     */
+    type GetIndexedField<T, K> =
+        // This covers a lot of ground: indexing arrays and strings by number,
```

Comment:
I think this case is fairly self explanatory. Just remove this comment since it's more distracting than helpful.